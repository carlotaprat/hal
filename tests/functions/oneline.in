def foo: print x; return bar

def bar: print y
