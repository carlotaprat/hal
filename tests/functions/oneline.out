digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FUNDEF"];
  n1 [label="FUNDEF"];
  n2 [label="foo"];
  n3 [label="PARAMS"];
  n4 [label="BLOCK"];
  n4 [label="BLOCK"];
  n5 [label="FUNCALL"];
  n5 [label="FUNCALL"];
  n6 [label="print"];
  n7 [label="ARGS"];
  n7 [label="ARGS"];
  n8 [label="FUNCALL"];
  n8 [label="FUNCALL"];
  n9 [label="x"];
  n10 [label="ARGS"];
  n11 [label="RETURN"];
  n11 [label="RETURN"];
  n12 [label="FUNCALL"];
  n12 [label="FUNCALL"];
  n13 [label="bar"];
  n14 [label="ARGS"];
  n15 [label="FUNDEF"];
  n15 [label="FUNDEF"];
  n16 [label="bar"];
  n17 [label="PARAMS"];
  n18 [label="BLOCK"];
  n18 [label="BLOCK"];
  n19 [label="FUNCALL"];
  n19 [label="FUNCALL"];
  n20 [label="print"];
  n21 [label="ARGS"];
  n21 [label="ARGS"];
  n22 [label="FUNCALL"];
  n22 [label="FUNCALL"];
  n23 [label="y"];
  n24 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "FUNDEF"
  n1 -> n2 // "FUNDEF" -> "foo"
  n1 -> n3 // "FUNDEF" -> "PARAMS"
  n1 -> n4 // "FUNDEF" -> "BLOCK"
  n4 -> n5 // "BLOCK" -> "FUNCALL"
  n5 -> n6 // "FUNCALL" -> "print"
  n5 -> n7 // "FUNCALL" -> "ARGS"
  n7 -> n8 // "ARGS" -> "FUNCALL"
  n8 -> n9 // "FUNCALL" -> "x"
  n8 -> n10 // "FUNCALL" -> "ARGS"
  n4 -> n11 // "BLOCK" -> "RETURN"
  n11 -> n12 // "RETURN" -> "FUNCALL"
  n12 -> n13 // "FUNCALL" -> "bar"
  n12 -> n14 // "FUNCALL" -> "ARGS"
  n0 -> n15 // "BLOCK" -> "FUNDEF"
  n15 -> n16 // "FUNDEF" -> "bar"
  n15 -> n17 // "FUNDEF" -> "PARAMS"
  n15 -> n18 // "FUNDEF" -> "BLOCK"
  n18 -> n19 // "BLOCK" -> "FUNCALL"
  n19 -> n20 // "FUNCALL" -> "print"
  n19 -> n21 // "FUNCALL" -> "ARGS"
  n21 -> n22 // "ARGS" -> "FUNCALL"
  n22 -> n23 // "FUNCALL" -> "y"
  n22 -> n24 // "FUNCALL" -> "ARGS"

}

